src/guardian/guardian.service.ts

```diff
@@ -136,14 +137,25 @@ export class GuardianService implements OnModuleInit {
         blockHash: blockData.blockHash,
       });
 
-      await Promise.all(
-        stakingModules.map(async (stakingRouterModule) => {
-          const stakingModuleData =
-            await this.stakingModuleGuardService.getStakingRouterModuleData(
-              stakingRouterModule,
-              blockHash,
-            );
+      // TODO: check only if one of nonce changed
+      const modulesIdWithDuplicateKeys: number[] =
+        this.stakingModuleGuardService.checkVettedKeysDuplicates(
+          stakingModulesData,
+          blockData,
+        );
+
+      const stakingModulesWithoutDuplicates: StakingModuleData[] =
+        this.stakingModuleGuardService.excludeModulesWithDuplicatedKeys(
```

Comment:
After we will do check duplication for each module separatly this would be reduntant

Reply:
i dont think check separatly duplicates is a good idea
we look for duplicates across modules